### DISCUSSION TITLE
Snapshot StatefulSet Volumes | Stash

### DISCUSSION THREAD

**Discussion by giscus** [appscode/community#57] [Stash]
# Snapshot StatefulSet Volumes | Stash

Snapshotting the volumes of a StatefulSet This guide will show you how to use Stash to snapshot the volumes of a StatefulSet and restore them from the snapshots using Kubernetes VolumeSnapshot API. In this guide, we are going to backup the volumes in Google Cloud Platform with the help of GCE Persistent Disk CSI Driver.
Before You Begin At first, you need to be familiar with the GCE Persistent Disk CSI Driver.

https://stash.run/docs/v2023.08.18/guides/volumesnapshot/statefulset/

**Comment by sudarshan-shapur-aera** [comment]
Is restoration on existing volumes supported?
i.e.
1) we have backed up the PVC's of STS 
2) Now imagine for some reason the PVC's become corrupted/ in a bad state.
3) Can we perform restore on this existing PVC's
